proxy/http2/Http2Stream.cc

```diff
@@ -675,6 +675,17 @@ Http2Stream::destroy()
   }
   chunked_handler.clear();
   super::destroy();
+
+  // Current Http2ConnectionState implementation uses a memory pool for instantiating streams and DLL<> stream_list for storing
+  // active streams. Destroying a stream before deleting it from stream_list and then creating a new one + reusing the same chunk
+  // from the memory pool right away always leads to destroying the DLL structure (deadlocks, inconsistencies).
+  // The following is meant as a safety net since the consequences are disastrous. Until the design/implementation changes it seems
+  // less error prone to (double) delete before destroying (noop if already deleted).
+  if (parent) {
+    static_cast<Http2ClientSession *>(parent)->connection_state.delete_stream(this);
+    Warning("Http2Stream was about to be deallocated without removing it from the active stream list");
```

Comment:
If the streams are removed properly from the active stream list before this catch-all-delete-stream call and freeing the stream's memory we should not see this at all.

With the current 6.2.1 code it took 1-3 days for this to occur so even if the problem persists (not deleting the streams on time) it seems reasonable to believe we would see the working with similar frequency (once in 1-3 days).